src/ffetch.js

```diff
@@ -9,4 +8,128 @@
  * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
- */
\ No newline at end of file
+ */
+
+async function* request(url, context) {
+  const { chunks, fetch } = context;
+  for (let offset = 0, total = Infinity; offset < total; offset += chunks) {
+    const resp = await fetch(`${url}?offset=${offset}&limit=${chunks}`);
+    if (resp.ok) {
+      const json = await resp.json();
+      total = json.total;
+      for (const entry of json.data) yield entry;
+    } else {
+      return;
```

Comment:
I can see the `.retry(3)` feature coming already :)

Reply:
we can add some error handling when we see the need for it.

Reply:
in addition, we can retry with [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff). Not a blocker though, can be added based on demand.